### ISSUE TITLE
Linux desktop app - cert warning for cdn.segment.com

### ISSUE THREAD

**Issue by derek-shnosh** [mattermost/desktop#1310]
I confirm (by marking "x" in the [ ] below: [x]):

- [x] This is not a troubleshooting question. [Troubleshooting questions go here: http://www.mattermost.org/troubleshoot/](http://www.mattermost.org/troubleshoot/).
- [x] This doesn't reproduce on web browsers (such as in Chrome). If it does, [issue reports go to the Mattermost Server repository](https://github.com/mattermost/platform/issues).
- [x] I have read [contributing guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md).

---

**Summary**
Cert warning when launching linux client when **Enable Diagnostics and Error Reporting** is enabled (System Console > Environment > Logging).

**Environment**
- Operating System: Kubuntu 20.04
- Mattermost Desktop App version: 4.4.2
- Mattermost Server version: 5.23.0

**Steps to reproduce**
- Server deployed with in VMware, TLS with LetsEncrypt enabled in config.json.
- Accessing via different browsers on different OSes does not reproduce the issue.
- Accessing via Windows desktop client does not reproduce this issue; don't have a Mac to test with.

**Expected behavior**
Launch Linux desktop app without cert warning.

**Observed behavior**
First error message...
![image](https://user-images.githubusercontent.com/21090563/83610631-9b438f80-a534-11ea-9cae-3c756444028c.png)

More details...
![image](https://user-images.githubusercontent.com/21090563/83610664-a8f91500-a534-11ea-8a5b-720458c5675d.png)

Clicking "Trust Insecure Certificate" loads the mattermost app, but its blank.
Clicking "Cancel Connection" loads the mattermost app and it operates normally.

**Possible fixes**
Disable Diagnostics and Error Reporting.

**Comment by amyblais** [comment]
@Willyfrog Is this related to https://mattermost.atlassian.net/browse/MM-25750?

**Comment by Willyfrog** [comment]
no, MM-25750 is linked to https://mattermost.atlassian.net/browse/MM-22648

in this case is an issue with the store not trusting the CA. Since it is linux, I'm not sure if this is chrome handling the valid CA list. if that's the case, the it would need to upgrade to a more recent version of electron to remove the warning. (does 4.5.0-rc1 have this behaviour?)

I'm more concerned that it throws a blank page, as once it is accepted it should work as usual, are there any other errors on the dev tools?

**Comment by amyblais** [comment]
@derek-shnosh Can you help check if you see the same issue on 4.5.0-rc1 and if there are any other errors in the developer tools?

**Comment by derek-shnosh** [comment]
@amyblais Sure thing, I just installed 4.5.0-rc1-6473 (over 4.4.2-6223) and I received the same error the first time I launched it.

After trusting the insecure cert, the app still loaded with a blank page but there was a **< Back** banner at the top that loaded the UI once clicked.

I closed the app and reopened it, did not receive the cert warning; however this was in the console...

![image](https://user-images.githubusercontent.com/21090563/83663936-2b59f700-a57e-11ea-87d3-a5640d63104d.png)

URL: `https://cdn.segment.com/analytics.js/v1/davohjASA4Ox1BYPUkDcOYzGVvuwI46V/analytics.min.js`

I reverted back to 4.4.2 and found that I can use `Alt-LeftArrow` after trusting the cert and it will load the UI (similarly to the **< Back** banner behavior in 4.5.0-rc1).

Here is the console message from 4.4.2...

![image](https://user-images.githubusercontent.com/21090563/83664435-d8347400-a57e-11ea-9dfe-fcaa64fd5bc8.png)

URL here is the same as one above.

**Comment by Willyfrog** [comment]
feels weird that you can't find that library, I can access the segment's CDN without much problem.
is there something intercepting the network call?

**Comment by derek-shnosh** [comment]
Pihole was blocking that URL from home (via default Gravity list), the URL loads fine from a different environment. I'm unsure if the desktop app is intended to show a blank white screen if that is unreachable, but at least that explains the 404 error.

I will test from Linux in this environment shortly.

**Comment by Willyfrog** [comment]
No, since it was having ssl problems, it reached a different bug https://mattermost.atlassian.net/browse/MM-25364

**Comment by derek-shnosh** [comment]
Alright, I am not experiencing the blank white window issue while using Linux in this different network environment (without pihole).

I did check the dev tools > console and noticed that its not trying `cdn.segment.com` though, instead it's going to `cdn.rudderlabs.com` and getting HTTP/500 status.

![image](https://user-images.githubusercontent.com/21090563/83815028-43ac3d80-a674-11ea-8470-cf066f4ec0e8.png)

Either way, I've disabled the *Diagnostics and Error Reporting* feature as my work around, and since you guys are already aware of the blank white screen issue, this can probably be closed.
